lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1482 @@
+    Start-ExternalCommand { git -C $projectDir log -10 --pretty=format:"%h - %an, %ae, %ar : %s" } `
+        -ErrorMessage "Failed to show git log."
+
+    $ret = echo "$ZuulRef" | Where-Object { $_ -match "^refs/tags/" }
+    if ($ret) {
+        Start-ExternalCommand {
+                git -C $projectDir fetch --tags "$ZuulUrl/$ZuulProject"
+                git -C $projectDir checkout $ZuulRef
+                git -C $projectDir reset --hard $ZuulRef
+            } `
+                    -ErrorMessage "Failed to fetch tags from: $ZuulUrl/$ZuulProject"
+    }
+    elseif (!$ZuulNewrev) {
+        Start-ExternalCommand {
+                git -C $projectDir fetch "$ZuulUrl/$ZuulProject" $ZuulRef
+                git -C $projectDir checkout FETCH_HEAD
+                git -C $projectDir reset --hard FETCH_HEAD
+            } `
+                    -ErrorMessage "Failed to fetch: $ZuulUrl/$ZuulProject $ZuulRef"
+    }
+    else {
+        Start-ExternalCommand {
+                git -C $projectDir checkout $ZuulNewrev
+                git -C $projectDir reset --hard $ZuulNewrev
+            } `
+                    -ErrorMessage "Failed to checkout $ZuulNewrev"
+    }
+
+    try {
+        Start-ExternalCommand { git -C $projectDir clean -x -f -d -q } -ErrorMessage "Failed to git clean"
+    } catch {
+        sleep 1
+        Start-ExternalCommand { git -C $projectDir clean -x -f -d -q } -ErrorMessage "Failed to git clean"
+    }
+
+    if (Test-Path $gitmodulesDir) {
+        Start-ExternalCommand { 
+                git -C $projectDir submodule init
+                git -C $projectDir submodule sync
+                git -C $projectDir submodule update --init
+            } `
+                    -ErrorMessage "Failed to init submodule"
+    }
+
+    echo "Final result:"
+    echo "Git branch output:"
+    Start-ExternalCommand { git -C $projectDir branch } -ErrorMessage "Failed to show git branch."
+    echo "Git log output:"
+    Start-ExternalCommand { git -C $projectDir log -10 --pretty=format:"%h - %an, %ae, %ar : %s" } `
+        -ErrorMessage "Failed to show git log."
+
+}
+
+
+function Install-Projects ($buildFor) {
+    # Shared for all projects
+    $projectDir = $buildFor -replace 'openstack/'
+    Write-JujuLog "Installing $projectDir"
+    $openstackBuild = Join-Path $BUILD_DIR "openstack"
+    Start-ExecuteWithRetry {
+        Install-OpenStackProjectFromRepo "$openstackBuild\$projectDir"
+    }
+    
+    # Individual projects
+    # Nova
+    if ($buildFor -eq "openstack/nova") {
+        Invoke-BuildNova
+    }
+    # Networking-Hyperv
+    if ($buildFor -eq "openstack/networking-hyperv") {
+        Invoke-BuildNeutron
+    }
+}
+
+
+function Invoke-BuildNova {
+    $novaBin = (Get-CharmServices)['nova']['binary']
+    if (!(Test-Path $novaBin)) {
+       Throw "$novaBin was not found."
+    }
+
+    Write-JujuLog "Copying default config files"
+    $defaultConfigFiles = @('rootwrap.d', 'api-paste.ini', 'cells.json',
```

Comment:
se above comment..this occurs when nova is installed, be it from master or as a prepped project